### PR TITLE
Add event sampling. Name threads.

### PR DESCRIPTION
**Pull request by drichelson**
Add event sampling. Name threads.

### CODE REVIEW

File: src/main/java/com/launchdarkly/client/LDConfig.java [hunk 1]
```diff
@@ -335,6 +339,21 @@ public Builder startWaitMillis(long startWaitMillis) {
...
+     * Enable event sampling. When set to the default of zero, sampling is disabled and all events
+     * are sent back to LaunchDarkly. When set to greater than zero, there is a 1 in
+     * <code>samplingInterval</code> chance events will be will be sent.
+     *
+     * <p>Example: if you want 5% sampling rate, set <code>samplingInterval</code> to 20.
+     *
+     * @param samplingInterval the sampling interval.
+     * @return the builder
+     */
+    public Builder samplingInterval(int samplingInterval) {
```
Comment (drichelson) [inline src/main/java/com/launchdarkly/client/LDConfig.java]: I copied the semantics in the go client, but putting myself in the user's shoes I prefer using a percentage value here (and everywhere else) in the form of a floating point value from 0 to 1.0 where at 1.0 all events get sent and at 0.0 none get sent. I think consistency is more important, so we should leave it or change it everywhere.

**Comment by drichelson** [comment]
Corrected target branch from #43

File: build.gradle [hunk 2]
```diff
@@ -6,18 +6,15 @@ apply plugin: 'idea'
...
 
 configurations.all {
-    // check for updates every build
+    // check for updates every build for dependencies with: 'changing: true'
     resolutionStrategy.cacheChangingModulesFor 0, 'seconds'
 }
 
 repositories {
-    mavenCentral()
     mavenLocal()
```
Comment (drichelson) [inline build.gradle]: order matters here, so we look first locally, then elsewhere.

File: build.gradle [hunk 3]
```diff
@@ -38,7 +35,7 @@ dependencies {
     compile "redis.clients:jedis:2.8.0"
     testCompile "org.easymock:easymock:3.4"
     testCompile 'junit:junit:[4.10,)'
-    testRuntime "org.slf4j:slf4j-simple:1.7.7"
+    testRuntime "ch.qos.logback:logback-classic:1.1.3"
```
Comment (drichelson) [inline build.gradle]: Using logback for tests allows for more helpful log messages that include source code  line number.

**Comment by jkodumal** [comment]
:+1:

**Pull request merged**